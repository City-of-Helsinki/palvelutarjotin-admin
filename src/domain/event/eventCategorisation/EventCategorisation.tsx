import * as React from 'react';
import { useTranslation } from 'react-i18next';

import TextTitle from '../../../common/components/textTitle/TextTitle';
import {
  EventQuery,
  LocalisedFieldsFragment,
} from '../../../generated/graphql';
import { Language } from '../../../types';
import getLocalizedString from '../../../utils/getLocalizedString';
import styles from './eventCategorisation.module.scss';

type Props = {
  eventData: EventQuery;
  language: Language;
};

const EventCategorisation: React.FC<Props> = ({ eventData, language }) => {
  const { t } = useTranslation();
  const neededOccurrences = eventData.event?.pEvent?.neededOccurrences;
  const inLanguage = eventData.event?.inLanguage;
  const audience = eventData.event?.audience;
  const keywords = eventData.event?.keywords;
  const additionalCriteria = eventData.event?.additionalCriteria;
  const categories = eventData.event?.categories;
  const offer = eventData.event?.offers?.[0];
  const price = offer?.price?.[language];
  const priceDescription = offer?.description?.[language];
  const isFree = offer?.isFree ?? !price;
  const priceInfoUrl = offer?.infoUrl?.[language];

  // Get keywords that are not already included in categories or activities
  const filteredKeywords = keywords?.filter(
    (keyword) =>
      !categories?.find((category) => category.id === keyword.id) &&
      !additionalCriteria?.find((activity) => activity.id === keyword.id)
  );

  const createLink = (prefix: string, url: string) => (
    <>
      {prefix}{' '}
      <a href={url} rel="noopener noreferrer" target="_blank">
        {url}
      </a>
    </>
  );

  const arrayToText = (
    items: { name?: LocalisedFieldsFragment | null }[] | undefined
  ) => {
    return (
      items
        ?.map((item) => getLocalizedString(item.name || {}, language))
        .filter((item) => item)
        .sort()
        .join(', ') || '-'
    );
  };

  return (
    <div className={styles.eventCategorisation}>
      <h2>{t('eventDetails.categorisation.title')}</h2>
      <div className={styles.infoRow}>
        <div>
          <TextTitle>
            {t('eventDetails.categorisation.labelInLanguage')}
          </TextTitle>
          {<p>{arrayToText(inLanguage)}</p>}
        </div>
        <div>
          <TextTitle>
            {t('eventDetails.categorisation.labelAudience')}
          </TextTitle>
          <p>{arrayToText(audience)}</p>
        </div>
      </div>

      <div className={styles.infoRow}>
        <div>
          <TextTitle>
            {t('eventDetails.categorisation.labelCategories')}
          </TextTitle>
          {<p>{arrayToText(categories)}</p>}
        </div>
        <div>
          <TextTitle>
            {t('eventDetails.categorisation.labelActivities')}
          </TextTitle>
          <p>{arrayToText(additionalCriteria)}</p>
        </div>
      </div>

      <TextTitle>{t('eventDetails.categorisation.labelKeywords')}</TextTitle>
      <p>{arrayToText(filteredKeywords)}</p>

      <div className={styles.infoRow}>
        <div>
          <TextTitle>{t('eventDetails.categorisation.labelPrice')}</TextTitle>
          {isFree && <p>{t('eventDetails.categorisation.eventIsFree')}</p>}
          {!isFree && (
            <p className={styles.eventPriceMultiline}>
              {price}
              {priceDescription && `\n${priceDescription}`}
              {priceInfoUrl && createLink(`\nInfo: `, priceInfoUrl)}
            </p>
          )}
        </div>
        <div>
          <TextTitle>
            {t('eventDetails.categorisation.labelNeededOccurrences')}
          </TextTitle>
          <p>{neededOccurrences}</p>
        </div>
      </div>
    </div>
  );
};

export default EventCategorisation;
