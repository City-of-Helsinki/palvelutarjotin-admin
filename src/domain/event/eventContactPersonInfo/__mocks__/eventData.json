{
  "event": {
    "pEvent": {
      "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
      "contactPerson": {
        "id": "UGVyc29uTm9kZTo0MGZmYTIwMS1mOWJhLTQyZTYtYjY3Ny01MWQyM2Q4OGQ4ZDk=",
        "emailAddress": "",
        "name": "Santeri",
        "phoneNumber": "",
        "language": "FI",
        "__typename": "PersonNode"
      },
      "contactEmail": "testi@testinen.fi",
      "contactPhoneNumber": "12345678",
      "organisation": {
        "id": "T3JnYW5pc2F0aW9uTm9kZTox",
        "name": "Kulttuurin ja vapaa-ajan toimiala",
        "persons": {
          "edges": [],
          "__typename": "PersonNodeConnection"
        },
        "phoneNumber": "123123123123",
        "publisherId": "ahjo:u480400",
        "type": "PROVIDER",
        "__typename": "OrganisationNode"
      },
      "__typename": "PalvelutarjotinEventNode"
    },
    "__typename": "Event"
  }
}
