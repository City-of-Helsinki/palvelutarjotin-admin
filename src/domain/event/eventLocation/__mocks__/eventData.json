{
  "event": {
    "id": "palvelutarjotin:afzunowba4",
    "internalId": "https://api.hel.fi/linkedevents-test/v1/event/palvelutarjotin:afzunowba4/",
    "name": {
      "en": null,
      "fi": "Tapahtuma 13.7.2020",
      "sv": null,
      "__typename": "LocalisedObject"
    },
    "shortDescription": {
      "en": null,
      "fi": "Tapahtuma 13.7.2020",
      "sv": null,
      "__typename": "LocalisedObject"
    },
    "description": {
      "en": null,
      "fi": "Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020 Tapahtuma 13.7.2020",
      "sv": null,
      "__typename": "LocalisedObject"
    },
    "images": [
      {
        "id": "48566",
        "internalId": "https://api.hel.fi/linkedevents-test/v1/image/48566/",
        "license": "cc_by",
        "name": "Screenshot 2020-07-07 at 10.48.19.png",
        "url": "https://api.hel.fi/linkedevents-test/media/images/Screenshot_2020-07-07_at_10.48.19.png",
        "cropping": "59,0,503,444",
        "photographerName": "asd",
        "altText": "asd",
        "__typename": "Image"
      }
    ],
    "infoUrl": {
      "en": null,
      "fi": "",
      "sv": null,
      "__typename": "LocalisedObject"
    },
    "pEvent": {
      "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
      "contactPerson": {
        "id": "UGVyc29uTm9kZTo0MGZmYTIwMS1mOWJhLTQyZTYtYjY3Ny01MWQyM2Q4OGQ4ZDk=",
        "emailAddress": "",
        "name": "Santeri",
        "phoneNumber": "",
        "language": "FI",
        "__typename": "PersonNode"
      },
      "contactEmail": "santtu_1993@hotmail.com",
      "contactPhoneNumber": "0405542959",
      "duration": 11,
      "enrolmentEndDays": 3,
      "enrolmentStart": "2020-07-13T06:00:00+00:00",
      "neededOccurrences": 3,
      "organisation": {
        "id": "T3JnYW5pc2F0aW9uTm9kZTox",
        "name": "Kulttuurin ja vapaa-ajan toimiala",
        "persons": {
          "edges": [
            {
              "node": {
                "id": "UGVyc29uTm9kZTpjZjNkN2NlYS0xNGQxLTQzNmQtOTNmZS0zOTI4Mzc2ZTQ2YTE=",
                "emailAddress": "admin@example.com",
                "name": "Admin",
                "phoneNumber": "",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTpiNTk5MThiNi1iOGRlLTRmOGItYjM3Yy0xYWRhYThmOTRkNWM=",
                "emailAddress": "jori.lindell@gmail.com",
                "name": "Jori",
                "phoneNumber": "04412345",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTo2MmYyOTY1ZC05NWEyLTRmODYtODJlOS02OWFhODdhZGVjNDY=",
                "emailAddress": "lucquocquyen@gmail.com",
                "name": "Khin Loc",
                "phoneNumber": "123123213",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTowM2Y4YTRiNS1kOTIzLTQ3NWQtOTBmYS1kYzIxY2MwMGM2MDY=",
                "emailAddress": "matti.saarto@gmail.com",
                "name": "Testaaja",
                "phoneNumber": "123123123",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTo0NjYwOTU4Ny1mOTExLTQ1ZjEtYjEyYy1lNWMzZTJiNWJiMjc=",
                "emailAddress": "pirjo.mattila@hel.fi",
                "name": "Pipe",
                "phoneNumber": "123123123",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTo0MGZmYTIwMS1mOWJhLTQyZTYtYjY3Ny01MWQyM2Q4OGQ4ZDk=",
                "emailAddress": "",
                "name": "Santeri",
                "phoneNumber": "",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTpmNDg5YWFiZC1kZTE4LTQ1M2YtODIxZC1hNjgzNGMxNzZhZTg=",
                "emailAddress": "tomppa.makinen@gmail.com",
                "name": "Tomi MÃ¤kinen",
                "phoneNumber": "+358401231234",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            },
            {
              "node": {
                "id": "UGVyc29uTm9kZTowMDVjOGIyNS0zN2YwLTRjNmEtYjZjMi1kYTBkYzI2YTQzY2I=",
                "emailAddress": "",
                "name": "Ville",
                "phoneNumber": "",
                "language": "FI",
                "__typename": "PersonNode"
              },
              "__typename": "PersonNodeEdge"
            }
          ],
          "__typename": "PersonNodeConnection"
        },
        "phoneNumber": "123123",
        "publisherId": "ahjo:u480400",
        "type": "PROVIDER",
        "__typename": "OrganisationNode"
      },
      "occurrences": {
        "edges": [
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6ODg=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "en", "name": "English", "__typename": "LanguageType" },
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-03T09:00:00+00:00",
              "endTime": "2020-08-03T09:30:00+00:00",
              "placeId": "",
              "seatsTaken": 0,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6ODk=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" },
                { "id": "en", "name": "English", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-04T09:00:00+00:00",
              "endTime": "2020-08-04T09:30:00+00:00",
              "placeId": "",
              "seatsTaken": 0,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6OTA=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" },
                { "id": "en", "name": "English", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-05T09:00:00+00:00",
              "endTime": "2020-08-05T09:30:00+00:00",
              "placeId": "",
              "seatsTaken": 0,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6OTE=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" },
                { "id": "en", "name": "English", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-06T09:00:00+00:00",
              "endTime": "2020-08-06T09:30:00+00:00",
              "placeId": "",
              "seatsTaken": 0,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6OTI=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" },
                { "id": "en", "name": "English", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-07T09:30:00+00:00",
              "endTime": "2020-08-07T10:30:00+00:00",
              "placeId": "",
              "seatsTaken": 10,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6OTM=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" },
                { "id": "en", "name": "English", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-10T09:30:00+00:00",
              "endTime": "2020-08-10T10:00:00+00:00",
              "placeId": "",
              "seatsTaken": 25,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6OTQ=",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 30,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "en", "name": "English", "__typename": "LanguageType" },
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-12T09:00:00+00:00",
              "endTime": "2020-08-12T10:00:00+00:00",
              "placeId": "",
              "seatsTaken": 25,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6MTA0",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 10,
              "minGroupSize": 5,
              "maxGroupSize": 10,
              "autoAcceptance": true,
              "languages": [
                { "id": "en", "name": "English", "__typename": "LanguageType" },
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-19T12:30:00+00:00",
              "endTime": "2020-08-19T13:30:00+00:00",
              "placeId": "",
              "seatsTaken": 10,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          },
          {
            "node": {
              "id": "T2NjdXJyZW5jZU5vZGU6MTEw",
              "pEvent": {
                "id": "UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjcw",
                "__typename": "PalvelutarjotinEventNode"
              },
              "amountOfSeats": 10,
              "minGroupSize": 3,
              "maxGroupSize": 5,
              "autoAcceptance": false,
              "languages": [
                { "id": "fi", "name": "Finnish", "__typename": "LanguageType" },
                { "id": "en", "name": "English", "__typename": "LanguageType" }
              ],
              "startTime": "2020-08-20T12:30:00+00:00",
              "endTime": "2020-08-20T13:30:00+00:00",
              "placeId": "",
              "seatsTaken": 5,
              "__typename": "OccurrenceNode"
            },
            "__typename": "OccurrenceNodeEdge"
          }
        ],
        "__typename": "OccurrenceNodeConnection"
      },
      "__typename": "PalvelutarjotinEventNode"
    },
    "inLanguage": [
      {
        "id": "fi",
        "internalId": "https://api.hel.fi/linkedevents-test/v1/language/fi/",
        "name": {
          "en": null,
          "fi": "suomi",
          "sv": null,
          "__typename": "LocalisedObject"
        },
        "__typename": "InLanguage"
      },
      {
        "id": "en",
        "internalId": "https://api.hel.fi/linkedevents-test/v1/language/en/",
        "name": {
          "en": null,
          "fi": "englanti",
          "sv": null,
          "__typename": "LocalisedObject"
        },
        "__typename": "InLanguage"
      }
    ],
    "audience": [],
    "keywords": [
      {
        "id": "yso:p4363",
        "name": {
          "en": "families",
          "fi": "perheet",
          "sv": "familjer",
          "__typename": "LocalisedObject"
        },
        "internalId": "https://api.hel.fi/linkedevents-test/v1/keyword/yso:p4363/",
        "__typename": "Keyword"
      },
      {
        "id": "yso:p6165",
        "name": {
          "en": "migrants",
          "fi": "maahanmuuttajat",
          "sv": "invandrare",
          "__typename": "LocalisedObject"
        },
        "internalId": "https://api.hel.fi/linkedevents-test/v1/keyword/yso:p6165/",
        "__typename": "Keyword"
      }
    ],
    "location": {
      "id": "tprek:15376",
      "internalId": "https://api.hel.fi/linkedevents-test/v1/place/tprek:15376/",
      "name": {
        "en": "Soukka Library",
        "fi": "Soukan kirjasto",
        "sv": "SÃ¶kÃ¶ bibliotek",
        "__typename": "LocalisedObject"
      },
      "streetAddress": {
        "en": "Soukantie 4",
        "fi": "Soukantie 4",
        "sv": "SÃ¶kÃ¶vÃ¤gen 4",
        "__typename": "LocalisedObject"
      },
      "addressLocality": {
        "en": "Espoo",
        "fi": "Espoo",
        "sv": "Esbo",
        "__typename": "LocalisedObject"
      },
      "telephone": {
        "en": null,
        "fi": "+358 9 8165 7736",
        "sv": null,
        "__typename": "LocalisedObject"
      },
      "__typename": "Place"
    },
    "venue": {
      "id": "tprek:15376",
      "hasClothingStorage": true,
      "hasSnackEatingPlace": true,
      "outdoorActivity": true,
      "translations": [
        {
          "languageCode": "FI",
          "description": "asdasdasd",
          "__typename": "VenueTranslationType"
        }
      ],
      "__typename": "VenueNode"
    },
    "startTime": "2020-07-13T05:51:05.761000Z",
    "publicationStatus": "public",
    "datePublished": null,
    "__typename": "Event"
  }
}
