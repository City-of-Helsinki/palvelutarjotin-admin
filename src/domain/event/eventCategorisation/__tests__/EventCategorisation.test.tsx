import { render, screen } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import React from 'react';

import eventData from '../__mocks__/eventData.json';
import { EventQuery } from '../../../../generated/graphql';
import EventCategorisation from '../EventCategorisation';

test('matches snapshot', () => {
  const { container } = render(
    <EventCategorisation eventData={eventData as EventQuery} language="fi" />
  );

  expect(container).toMatchSnapshot();
});

test('renders ands display information correctly', () => {
  render(
    <EventCategorisation eventData={eventData as EventQuery} language="fi" />
  );

  // titles
  expect(screen.queryByText('Tapahtuma on ilmainen')).toBeVisible();
  expect(
    screen.queryByRole('heading', { name: 'Tapahtuman luokittelut' })
  ).toBeVisible();
  expect(screen.queryByText('Tapahtuman kielet')).toBeVisible();
  expect(screen.queryByText('Tapahtuman avainsanat')).toBeVisible();
  expect(screen.queryByText('Hinta, €')).toBeVisible();
  expect(screen.queryByText('Tarvittavat käyntikerrat')).toBeVisible();

  // data
  expect(screen.queryByText('englanti, suomi')).toBeVisible();
  expect(screen.queryByText('maahanmuuttajat, perheet')).toBeVisible();
  expect(screen.queryByText('Tapahtuma on ilmainen')).toBeVisible();
  expect(screen.queryByText('3')).toBeVisible();
  expect(screen.queryByText('-')).toBeInTheDocument();
});

test('show correct texts with different data', () => {
  const newEventData = cloneDeep(eventData);
  newEventData.event.keywords.push({
    name: {
      fi: 'eläkeläiset',
    },
  } as any);
  newEventData.event.pEvent.neededOccurrences = 4;
  newEventData.event.inLanguage.push({
    name: {
      fi: 'ruotsi',
    },
  } as any);
  (newEventData.event.audience as any).push({
    name: {
      fi: 'eläkeläiset',
    },
  } as any);
  render(
    <EventCategorisation eventData={newEventData as EventQuery} language="fi" />
  );

  expect(
    screen.queryByText('eläkeläiset, maahanmuuttajat, perheet')
  ).toBeVisible();
  expect(screen.queryByText('4')).toBeVisible();
  expect(screen.queryByText('englanti, ruotsi, suomi')).toBeVisible();
  expect(screen.queryByText('eläkeläiset')).toBeVisible();
  expect(screen.queryByText('-')).not.toBeInTheDocument();
});
