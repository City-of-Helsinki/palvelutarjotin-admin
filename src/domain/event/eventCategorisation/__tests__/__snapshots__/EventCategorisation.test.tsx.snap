// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="eventCategorisation"
  >
    <h2>
      Tapahtuman luokittelut
    </h2>
    <div
      class="infoRow"
    >
      <div>
        <div
          class="textTitle"
        >
          Tapahtuman kielet
        </div>
        <p>
          englanti, suomi
        </p>
      </div>
      <div>
        <div
          class="textTitle"
        >
          Kohderyhmät
        </div>
        <p>
          -
        </p>
      </div>
    </div>
    <div
      class="infoRow"
    >
      <div>
        <div
          class="textTitle"
        >
          Kategoriat
        </div>
        <p>
          Musiikki, Teatteri
        </p>
      </div>
      <div>
        <div
          class="textTitle"
        >
          Aktiviteetit
        </div>
        <p>
          Luontokoulu, Työpaja
        </p>
      </div>
    </div>
    <div
      class="textTitle"
    >
      Tapahtuman avainsanat
    </div>
    <p>
      maahanmuuttajat, perheet
    </p>
    <div
      class="infoRow"
    >
      <div>
        <div
          class="textTitle"
        >
          Hinta, €
        </div>
        <p>
          Tapahtuma on ilmainen
        </p>
      </div>
      <div>
        <div
          class="textTitle"
        >
          Tarvittavat käyntikerrat
        </div>
        <p>
          3
        </p>
      </div>
    </div>
  </div>
</div>
`;
